Library/Homebrew/utils/diskusage.rb

```diff
@@ -0,0 +1,57 @@
+require "find"
+
+module Utils
+  class DiskUsage
+    attr_reader :size_in_bytes, :file_count
+
+    def initialize(path, excludes = [])
+      @size_in_bytes = 0
+      @file_count = 0
+      @path = path
+      @excludes = [".DS_Store"] + excludes
+      begin
+        Find.find(@path).each do |f|
+          if !File.directory?(f) and !f.match(/#{@excludes.join("|")}/)
```

Comment:
It's better to put `/#{@excludes.join("|")}/` outside of loop for better performance.

I would also be interesting to see some benchmark between ruby version with original shellout approach.